filters/f_auto_filters.c

```diff
@@ -89,28 +88,31 @@ static void deint_process(struct mp_filter *f)
     bool has_filter = true;
     if (img->imgfmt == IMGFMT_VDPAU) {
         char *args[] = {"deint", "yes",
+                        "interlaced-only", opts->deinterlace ? "no" : "yes",
                         "parity", field_parity, NULL};
         p->sub.filter =
             mp_create_user_filter(f, MP_OUTPUT_CHAIN_VIDEO, "vdpaupp", args);
     } else if (img->imgfmt == IMGFMT_D3D11) {
         char *args[] = {"deint", "yes",
-                        "interlaced-only", "yes",
+                        "interlaced-only", opts->deinterlace ? "no" : "yes",
                         "parity", field_parity, NULL};
         p->sub.filter =
             mp_create_user_filter(f, MP_OUTPUT_CHAIN_VIDEO, "d3d11vpp", args);
     } else if (img->imgfmt == IMGFMT_CUDA) {
         char *args[] = {"mode", "send_field",
+                        "deint", opts->deinterlace ? "0" : "1",
```

Comment:
shouldn't this also be `interlaced-only`?

Reply:
No, for cuda we insert bwdif_cuda which has a different syntax https://ffmpeg.org/ffmpeg-all.html#bwdif_005fcuda

Reply:
ok